src/H5FDs3comms.c

```diff
@@ -1304,23 +1315,24 @@ H5FD_s3comms_s3r_read(s3r_t *handle, haddr_t offset, size_t len, void *dest)
     else {
         /* authenticate request
          */
-        char authorization[512 + 1];
-        /*   512 := approximate max length...
-         *    67 <len("AWS4-HMAC-SHA256 Credential=///s3/aws4_request,"
-         *           "SignedHeaders=,Signature=")>
-         * +   8 <yyyyMMDD>
-         * +  64 <hex(sha256())>
-         * + 128 <max? len(secret_id)>
-         * +  20 <max? len(region)>
-         * + 128 <max? len(signed_headers)>
+        char authorization[2048 + 1];
```

Comment:
When we went to clean up warnings in HDF5 in recent times, one of the things responsible for several warnings was too large of stack-allocated buffers. Before this change the buffers were probably small enough at 513 bytes that they didn't cause warnings, but at 2049 bytes they likely will. For buffers of this size, the code here will probably need to change to using dynamic allocation to avoid warnings. We don't currently run into this in our -Werror builds because they don't build the ROS3 VFD.

Reply:
Oke. I have change memory allocation of the variable `authorization`, `buffer` and `signed_headers` to a dynamic one.

Reply:
Change the implementation such that `authorization`, `buffer1` and `signed_headers` are now allocated dynamically.